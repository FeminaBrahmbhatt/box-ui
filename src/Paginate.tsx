import React from "react";
import { PaginationProps } from "./@interfaces";
import classNames from "classnames";
import { Pagination } from "react-headless-pagination";
import { FiArrowLeft, FiArrowRight } from "react-icons/fi";

export const Paginate = ({
  totalPages,
  page,
  isMobile,
  className,
  setPage,
}: PaginationProps) => {
  const handlePageChange = (page: number) => {
    setPage(page);
  };

  const ArrowLeft = FiArrowLeft as React.FC<any>;
  const ArrowRight = FiArrowRight as React.FC<any>;

  if (isMobile) {
    return (
      <div className={classNames("flex w-full h-10 items-center", className)}>
        <ArrowLeft
          data-testid="prevBtn"
          size={20}
          className={classNames("mr-3 text-gray-500 dark:text-white", {
            "cursor-pointer": page !== 0,
            "opacity-50": page === 0,
          })}
          onClick={() => {
            if (page > 0) {
              handlePageChange(page - 1);
            }
          }}
        />
        <div className="flex justify-center flex-grow text-sm text-gray-700 select-none dark:text-white">
          {`Page ${page} of ${totalPages}`}
        </div>
        <ArrowRight
          data-testid="nextBtn"
          size={20}
          className={classNames("ml-3 text-gray-500 dark:text-white", {
            "cursor-pointer": page !== totalPages - 1,
            "opacity-50": page === totalPages - 1,
          })}
          onClick={() => {
            if (page < totalPages - 1) {
              handlePageChange(page + 1);
            }
          }}
        />
      </div>
    );
  }

  return (
    <Pagination
      className={classNames(
        "flex w-full items-center h-10 text-sm select-none",
        className
      )}
      currentPage={page}
      totalPages={totalPages}
      edgePageCount={2}
      setCurrentPage={handlePageChange}
      truncableText="..."
      middlePagesSiblingCount={2}
      truncableClassName="w-10 px-0.5 text-center dark:text-gray-500"
    >
      <Pagination.PrevButton
        className={classNames(
          "h-10 font-medium flex items-center mr-2 text-gray-500 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none",
          {
            "cursor-pointer": page !== 0,
            "opacity-50": page === 0,
          }
        )}
      >
        <ArrowLeft size={20} className={classNames("mr-3")} />
        Previous
      </Pagination.PrevButton>
      <div className="flex items-center justify-center flex-grow list-none">
        <Pagination.PageButton
          activeClassName="bg-primary-50 dark:bg-opacity-0 text-primary-600 dark:text-white"
          inactiveClassName="text-gray-500"
          className="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium"
        />
      </div>
      <Pagination.NextButton
        className={classNames(
          "h-10 font-medium flex items-center mr-2 text-gray-500 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none",
          {
            "cursor-pointer": page !== totalPages - 1,
            "opacity-50": page === totalPages - 1,
          }
        )}
      >
        Next
        <ArrowRight size={20} className={classNames("ml-3")} />
      </Pagination.NextButton>
    </Pagination>
  );
};
