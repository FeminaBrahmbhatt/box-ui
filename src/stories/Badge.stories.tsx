import React from 'react';
import {BadgeProps} from '../@interfaces';
import { Badge } from '../Badge';
import {images} from '../data/images'
import {Meta, StoryObj} from '@storybook/react';
import { StoryLayout } from '../StoryLayout';
import { FiArrowRight, FiStar } from 'react-icons/fi';
import { IconWrapper } from '../utils/IconWrapper';

export default {
    title: 'Example/Badge',
    component: Badge,
    parameters: {
        layout: 'centered',
    },
    tags: ['autodocs'],
} as Meta;
interface Props extends BadgeProps {
    darkMode: boolean;
}

export const Default:StoryObj<Props> = {
    args: {
        darkMode: false,
        variant: "success",
        size: "md",
        children:<span>Label</span>
    }
}

export const LeadingIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "success",
        size: "md",
        LeadingIcon: <IconWrapper Icon={FiStar} data-testid="leadingIcon"/>
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Label
                </Badge>
            </StoryLayout>
        )
    }
}

export const ImageIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "gray",
        size: "md",
        LeadingIcon: <img src={images.NL} alt="nl" className="w-4 h-4" data-testid="iconOnly" />
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Newzeland
                </Badge>
            </StoryLayout>
        )
    }
}

export const TrailingIconBadge:StoryObj<Props>  = {
    args: {
        darkMode: false,
        variant: "primary",
        size: "md",
        TrailingIcon: <IconWrapper Icon={FiArrowRight} data-testid="trailingIcon"/>
    },
    render: (args) => {
        return (
            <StoryLayout {...args} className="inline-flex flex-col space-y-2">
                <Badge {...args}>
                    Label
                </Badge>
            </StoryLayout>
        )
    }
}

