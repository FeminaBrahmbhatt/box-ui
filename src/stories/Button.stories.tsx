import React from "react";
import type { Meta, StoryObj } from "@storybook/react";
import { Button } from "../Button";
import { ButtonProps } from "../@interfaces";
import { FiArrowLeft, FiArrowRight, FiStar } from "react-icons/fi";
import { StoryLayout } from "../StoryLayout";
import { IconWrapper } from "../utils/IconWrapper";

export default{
  title: "Example/Button",
  component: Button,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} as Meta;
interface Props extends ButtonProps {
  darkMode: boolean;
}

type Story = StoryObj<Props>;

export const Default: Story = {
  args: {
    darkMode: false,
    children: "Button",
    size: 'md',
    variant: 'primary',
    disabled: false,
    LeadingIcon: undefined,
    TrailingIcon: undefined,
    IconOnly: undefined,
    className: ''
  },
  render: (args) => {
    return (
      <StoryLayout {...args} className="flex inline-flex flex-row space-x-2">
        <Button {...args}>Primary</Button>
      </StoryLayout>
    );
  },
};

export const Variants: Story = {
  args: {
    darkMode: false,
    variant: "primary",
  },
  render: (args) => {
    return (
      <StoryLayout {...args} className="flex inline-flex flex-row space-x-2">
        <Button variant="primary">Primary</Button>
        <Button variant="secondary">Secondary</Button>
        <Button variant="secondaryGray">Secondary Gray</Button>
        <Button variant="tertiary">Tertiary</Button>
        <Button variant="tertiaryGray">Tertiary Gray</Button>
      </StoryLayout>
    );
  },
};

export const Sizes: Story = {
  args: {
    darkMode: false,
    variant: "primary",
    size: "md",
  },
  render: (args) => {
    return (
      <StoryLayout {...args} className="flex inline-flex flex-row space-x-2">
        <Button size="sm">sm</Button>
        <Button size="md">md</Button>
        <Button size="lg">lg</Button>
        <Button size="xl">xl</Button>
        <Button size="2xl">2xl</Button>
      </StoryLayout>
    );
  },
};

export const TrailingIconButton: Story = {
  args: {
    darkMode: false,
    children: "Button",
    TrailingIcon: <IconWrapper Icon={FiArrowRight} data-testid="trailingIcon"/>,
  },  
  render: (args) => (
    <StoryLayout {...args}>
      <Button {...args} />
    </StoryLayout>
  ),
};

export const LeadingIconButton: Story = {
  args: {
    darkMode: false,
    children: "Button",
    LeadingIcon: <IconWrapper Icon={FiArrowLeft} data-testid="LeadingIcon"/>,
  },
  render: (args) => (
    <StoryLayout {...args}>
      <Button {...args} />
    </StoryLayout>
  ),
};

export const IconOnlyButton: Story = {
  args: {
    darkMode: false,
    IconOnly: <IconWrapper Icon={FiStar} data-testid="iconOnly"/>,
  },
  render: (args) => (
    <StoryLayout {...args}>
      <Button {...args} />
    </StoryLayout>
  ),
};
