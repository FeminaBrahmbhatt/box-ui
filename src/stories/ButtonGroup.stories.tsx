import React, { useState } from "react";
import { Meta, StoryObj } from "@storybook/react";
import { ButtonGroupProps, ButtonGroup } from "../ButtonGroup";
import { StoryLayout } from "../StoryLayout";
import { options1, options2, options3 } from "../data/options";

export default {
  title: "Example/ButtonGroup",
  component: ButtonGroup,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} as Meta;
interface Props extends ButtonGroupProps {
  darkMode: boolean;
}

const LabelButtonGroupComponent = (args: Props) => {
  const [activeItem1, setActiveItem1] = useState<string>(options1[1].value);

  return (
    <StoryLayout {...args}>
      <ButtonGroup
        {...args}
        options={options1}
        activeOption={activeItem1}
        setActiveOption={setActiveItem1}
      />
    </StoryLayout>
  );
};

export const LabelButtonGroup: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) =>  <LabelButtonGroupComponent {...args} />,
};

const IconButtonGroupComponent = (args: Props) => {
  const [activeView, setViewOption] = useState<string>("list");

  return (
    <StoryLayout {...args}>
      <ButtonGroup
        {...args}
        options={options3}
        activeOption={activeView}
        setActiveOption={setViewOption}
      />
    </StoryLayout>
  );
};

export const IconButtonGroup: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => <IconButtonGroupComponent {...args} />,
};

const LabelIconButtonGroupComponent = (args: Props) => {
  const [activeItem2, setActiveItem2] = useState<string>(options2[1].value);

  return (
    <StoryLayout {...args}>
      <ButtonGroup
        {...args}
        options={options2}
        activeOption={activeItem2}
        setActiveOption={setActiveItem2}
      />
    </StoryLayout>
  );
};

export const LabelIconButtonGroup: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => <LabelIconButtonGroupComponent {...args} />,
};
