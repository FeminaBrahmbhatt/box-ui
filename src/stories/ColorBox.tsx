import React from "react";
import { IColor } from "./@interfaces";

export const ColorBox = ({ code, hex, bgClass }: IColor) => (
<div
    key={bgClass}
    className="flex flex-col rounded-lg shadow-lg w-26 h-39 dark:bg-white">
      <div className={`h-20 ${bgClass} rounded-t-lg`} />
      <div className="mx-3 my-3">
          {code}
          <br/>
          {hex}
      </div>
    </div>
);