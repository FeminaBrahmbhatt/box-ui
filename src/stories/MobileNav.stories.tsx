import React, { useState } from "react";
import { Meta, StoryObj } from "@storybook/react";
import { MobileNavProps } from "../@interfaces";
import { SideNav } from "../Nav/SideNav";
import { StoryLayout } from "../StoryLayout";
import { navItemsTop, navItemsBottom } from "../data/navItems";
import { MobileNavbar } from "../Nav/MobileNav";
import { FiX } from "react-icons/fi"; // Directly import FiX
import type { IconBaseProps } from "react-icons";

const CloseIcon = FiX as React.FC<IconBaseProps>;
// Wrapper to fix TS JSX issues
 
export default {
  title: "Example/MobileNav",
  component: MobileNavbar,
} as Meta;

interface Props extends MobileNavProps {
  darkMode: boolean;
}

const MobileNavWithState = (args: Props) => {
  const [open, setOpen] = useState<boolean>(false);

  const handleToggle = () => {
    setOpen(!open);
  };

  return (
    <StoryLayout {...args}>
      <MobileNavbar open={open} toggleOpen={handleToggle} />

      {open && (
        <div
          data-testid="backdrop"
          className="fixed top-0 left-0 z-40 w-screen h-screen bg-gray-500 cursor-pointer bg-opacity-80"
          onClick={() => setOpen(false)}
        >
           <CloseIcon
            data-testid="close-icon"
            size={40}
            className="fixed text-white top-5 right-9"
            role="button"
            aria-label="Close sidebar"
          />
        </div>
      )}

      {open && (
        <SideNav
          className="relative z-50 h-screen -mt-20"
          navItemsTop={navItemsTop}
          navItemsBottom={navItemsBottom}
          open={true}
          setOpen={handleToggle}
          username="Veronica Woods"
          email="veronica@example.com"
        />
      )}
    </StoryLayout>
  );
};

export const Default: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => <MobileNavWithState {...args} />,
};
