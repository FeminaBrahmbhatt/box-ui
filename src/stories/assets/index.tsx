
export type CountryISO = "BE" | "DE" | "FR" | "IT" | "NL" | "RU" | "US";

type image = CountryISO;

// Development images
const devImages: Record<string, image> = {
  NL: require("./nl.svg")
};

// Production images (CDN)
// const prodImages: Record<string, image> = {
//   NL: "https://res.cloudinary.com/tailwindcss/image/upload/v1635279277/nl_tpy2ab.svg",
// };

export const assets = devImages;