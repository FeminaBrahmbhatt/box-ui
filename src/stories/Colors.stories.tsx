import React from "react";
import { Meta, StoryObj } from "@storybook/react";
import { ColorBox } from "../ColorBox";
import { colors } from "../data/colors";
import { StoryLayout } from "../StoryLayout";

export default {
  title: "Colors",
  parameters: {
    controls: { expanded: true },
  },
} as Meta;

interface Props {
  darkMode: boolean;
}

export const Default: StoryObj<Props> = {
  args: {
    darkMode: false,
  },
  render: (args) => (
    <StoryLayout {...args} className="grid grid-cols-2 gap-4 md:grid-cols-4 lg:grid-cols-6 xl:grid-cols-11">
      {colors.map((color) => (
        <ColorBox key={color.bgClass} code={color.code} hex={color.hex} bgClass={color.bgClass}/>
      ))}
    </StoryLayout>
  ),
};
