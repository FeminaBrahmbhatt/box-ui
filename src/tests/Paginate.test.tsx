import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import * as stories from '../stories/Paginate.stories';

const {
  DesktopPagination,
  MobilePagination
} = composeStories(stories);

describe('ðŸ“˜ Pagination Storybook Stories', () => {
  test('renders DesktopPagination component', () => {
    const Story = DesktopPagination;
    const { container } = render(
      Story(Story.args ?? {})
    );
    
    expect(screen.getByText('1')).toBeInTheDocument();
    expect(screen.getByText('Previous')).toBeInTheDocument();
    expect(screen.getByText('Next')).toBeInTheDocument();
    
    expect(container).toMatchSnapshot();
  });

  test('renders MobilePagination component', () => {
    const Story = MobilePagination;
    const { container } = render(
      Story(Story.args ?? {})
    );
    
    expect(container.querySelector('svg')).toBeInTheDocument(); // FiArrowRight

    expect(screen.getByTestId('prevBtn')).toBeInTheDocument();
    expect(screen.getByTestId('nextBtn')).toBeInTheDocument();    
    expect(screen.getByText('Page 0 of 10')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  test('renders DesktopPagination with total pages', () => {
    const Story = DesktopPagination;
    const { container } = render(
        Story({ totalPages: 5 })
    );
    
    for (let i = 1; i <= 5; i++) {
      expect(screen.getByText(i.toString())).toBeInTheDocument();
    }

    expect(container).toMatchSnapshot();
  });

  test('renders MobilePagination with page information', () => {
    const Story = MobilePagination;
    const { container } = render(
      Story({ totalPages: 5 })
    );
    
    expect(screen.getByText('Page 0 of 5')).toBeInTheDocument();
    
    expect(container).toMatchSnapshot();
  });

  test('renders DesktopPagination in dark mode', () => {
    const Story = DesktopPagination;
    const { container } = render(
      Story({ darkMode: true, totalPages: 5 })
    );

    expect(container.firstChild).toHaveClass('dark'); // Assuming you have dark mode styling applied with a class 'dark'

    expect(container).toMatchSnapshot();
  });

  test('renders MobilePagination in dark mode', () => {
    const Story = MobilePagination;
    const { container } = render(
      Story({ darkMode: true, totalPages: 5 })
    );

    expect(container.firstChild).toHaveClass('dark');

    expect(container).toMatchSnapshot();
  });
});
