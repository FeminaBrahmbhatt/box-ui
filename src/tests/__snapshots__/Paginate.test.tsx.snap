// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ“˜ Pagination Storybook Stories renders DesktopPagination component 1`] = `
<div>
  <div
    class="-m-4"
  >
    <div
      class="p-4"
    >
      <div
        class="flex w-full items-center h-10 text-sm select-none"
      >
        <button
          class="h-10 font-medium flex items-center mr-2 text-gray-500 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none opacity-50"
          disabled=""
          tabindex="-1"
        >
          <svg
            class="mr-3"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="19"
              x2="5"
              y1="12"
              y2="12"
            />
            <polyline
              points="12 19 5 12 12 5"
            />
          </svg>
          Previous
        </button>
        <div
          class="flex items-center justify-center flex-grow list-none"
        >
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium bg-primary-50 dark:bg-opacity-0 text-primary-600 dark:text-white"
              tabindex="0"
            >
              1
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              2
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              3
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              4
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              5
            </a>
          </li>
          <li
            class="w-10 px-0.5 text-center dark:text-gray-500"
          >
            ...
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              9
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              10
            </a>
          </li>
        </div>
        <button
          class="h-10 font-medium flex items-center mr-2 text-gray-500 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none cursor-pointer"
          tabindex="0"
        >
          Next
          <svg
            class="ml-3"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
            <polyline
              points="12 5 19 12 12 19"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ðŸ“˜ Pagination Storybook Stories renders DesktopPagination in dark mode 1`] = `
<div>
  <div
    class="dark bg-gray-900 -m-4"
  >
    <div
      class="p-4"
    >
      <div
        class="flex w-full items-center h-10 text-sm select-none"
      >
        <button
          class="h-10 font-medium flex items-center mr-2 text-gray-500 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none opacity-50"
          disabled=""
          tabindex="-1"
        >
          <svg
            class="mr-3"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="19"
              x2="5"
              y1="12"
              y2="12"
            />
            <polyline
              points="12 19 5 12 12 5"
            />
          </svg>
          Previous
        </button>
        <div
          class="flex items-center justify-center flex-grow list-none"
        >
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium bg-primary-50 dark:bg-opacity-0 text-primary-600 dark:text-white"
              tabindex="0"
            >
              1
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              2
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              3
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              4
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              5
            </a>
          </li>
        </div>
        <button
          class="h-10 font-medium flex items-center mr-2 text-gray-500 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none cursor-pointer"
          tabindex="0"
        >
          Next
          <svg
            class="ml-3"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
            <polyline
              points="12 5 19 12 12 19"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ðŸ“˜ Pagination Storybook Stories renders DesktopPagination with total pages 1`] = `
<div>
  <div
    class="-m-4"
  >
    <div
      class="p-4"
    >
      <div
        class="flex w-full items-center h-10 text-sm select-none"
      >
        <button
          class="h-10 font-medium flex items-center mr-2 text-gray-500 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none opacity-50"
          disabled=""
          tabindex="-1"
        >
          <svg
            class="mr-3"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="19"
              x2="5"
              y1="12"
              y2="12"
            />
            <polyline
              points="12 19 5 12 12 5"
            />
          </svg>
          Previous
        </button>
        <div
          class="flex items-center justify-center flex-grow list-none"
        >
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium bg-primary-50 dark:bg-opacity-0 text-primary-600 dark:text-white"
              tabindex="0"
            >
              1
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              2
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              3
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              4
            </a>
          </li>
          <li>
            <a
              class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer font-medium text-gray-500"
              tabindex="0"
            >
              5
            </a>
          </li>
        </div>
        <button
          class="h-10 font-medium flex items-center mr-2 text-gray-500 dark:text-white hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none cursor-pointer"
          tabindex="0"
        >
          Next
          <svg
            class="ml-3"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
            <polyline
              points="12 5 19 12 12 19"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ðŸ“˜ Pagination Storybook Stories renders MobilePagination component 1`] = `
<div>
  <div
    class="-m-4"
  >
    <div
      class="p-4"
    >
      <div
        class="flex w-full h-10 items-center"
      >
        <svg
          class="mr-3 text-gray-500 dark:text-white opacity-50"
          data-testid="prevBtn"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="19"
            x2="5"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 19 5 12 12 5"
          />
        </svg>
        <div
          class="flex justify-center flex-grow text-sm text-gray-700 select-none dark:text-white"
        >
          Page 0 of 10
        </div>
        <svg
          class="ml-3 text-gray-500 dark:text-white cursor-pointer"
          data-testid="nextBtn"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="5"
            x2="19"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 5 19 12 12 19"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`ðŸ“˜ Pagination Storybook Stories renders MobilePagination in dark mode 1`] = `
<div>
  <div
    class="dark bg-gray-900 -m-4"
  >
    <div
      class="p-4"
    >
      <div
        class="flex w-full h-10 items-center"
      >
        <svg
          class="mr-3 text-gray-500 dark:text-white opacity-50"
          data-testid="prevBtn"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="19"
            x2="5"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 19 5 12 12 5"
          />
        </svg>
        <div
          class="flex justify-center flex-grow text-sm text-gray-700 select-none dark:text-white"
        >
          Page 0 of 5
        </div>
        <svg
          class="ml-3 text-gray-500 dark:text-white cursor-pointer"
          data-testid="nextBtn"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="5"
            x2="19"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 5 19 12 12 19"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`ðŸ“˜ Pagination Storybook Stories renders MobilePagination with page information 1`] = `
<div>
  <div
    class="-m-4"
  >
    <div
      class="p-4"
    >
      <div
        class="flex w-full h-10 items-center"
      >
        <svg
          class="mr-3 text-gray-500 dark:text-white opacity-50"
          data-testid="prevBtn"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="19"
            x2="5"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 19 5 12 12 5"
          />
        </svg>
        <div
          class="flex justify-center flex-grow text-sm text-gray-700 select-none dark:text-white"
        >
          Page 0 of 5
        </div>
        <svg
          class="ml-3 text-gray-500 dark:text-white cursor-pointer"
          data-testid="nextBtn"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="5"
            x2="19"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 5 19 12 12 19"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
